client/packages/common/src/ui/forms/JsonForms/useJsonForms.tsx

```diff
@@ -123,6 +123,10 @@ export const useJsonForms = (
     }
   }, [initialData, data, isSaving, isLoading]);
 
+  useEffect(() => {
+    setData(initialData);
+  }, [initialData]);
```

Comment:
It seems to work but why? isn't the modal loaded all the time now? hwy is initialData changing then and reopening the modal?

Reply:
yes, the modal is loaded all the time. `initialData` won't change on every open of the modal - but it can now change if the `docName` is changed - as that is a prop on `useJsonForms` that hook will re-render; that will call `useDocumentLoader` which was previously keeping data in local state and simply returning the same object. Now that the second hook isn't keeping data locally, it returns the object related to the `docName` ( which itself might be a cached version from react-query 🙄 but will be related to the docName so we're ok! )

Reply:
seems to work even if docName stays the same, e.g. clicking the modal away and then reopening it, confused...